src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -158,7 +158,7 @@ public async Task<uint256> SendTransactionAsync(string hex)
         /// <returns>The new address.</returns>
         [ActionName("getnewaddress")]
         [ActionDescription("Returns a new wallet address for receiving payments.")]
-        public NewAddressModel GetNewAddress(string account, string addressType)
+        public NewAddressModel GetNewAddress(string account = "", string addressType = "")
```

Comment:
Should we have an optional parameter to indicate the flow generating new or asking for a current unused?

Reply:
Or perhaps this should be a setting value that defaults to false?

Reply:
We should keep this compatible with how other Bitcoin implementations work. Adding additional parameters and/or configuration settings would require the integrating party to do additional work. That will be a hurdle for exchanges to adopt the coins because every new exchange has to write custom code or has to have custom knowledge to support our coins.

Reply:
The problem this is an HD wallet and just generating a new address comes with consequences (one needs to maintain a standard gap for recovery to work)

That's why I propose a settings flag to invert this, alternatively we can pass a mandatory field in recovery that takes a gap then an exchange that does recovery can juts put a very big number there

Reply:
Settings flag already in place: "walletaddressbuffer" defaults to 20